src/backend/tests/work-packages.test.ts

```diff
@@ -0,0 +1,54 @@
+import request from 'supertest';
+import express from 'express';
+import workPackageRouter from '../src/routes/work-packages.routes';
+import { WBS_Element_Status } from '@prisma/client';
+import prisma from '../src/prisma/prisma';
+
+const app = express();
+app.use(express.json());
+app.use('/', workPackageRouter);
+
+// jest.mock('../src/utils/work-packages.utils');
+
+const createWorkPackagePayload = {
+  projectWbsNum: '1.2.0',
+  name: 'Pack your bags',
+  crId: 2,
+  userId: 1,
+  startDate: new Date('01/01/21'),
+  duration: 5,
+  dependencies: [
+    {
+      wbsElementId: 65,
+      dateCreated: new Date('12/25/20'),
+      carNumber: 1,
+      projectNumber: 1,
+      workPackageNumber: 1,
+      name: 'prereq',
+      status: WBS_Element_Status.COMPLETE
+    }
+  ],
+  expectedActivities: [{ descriptionId: 5, dateAdded: '01/05/21', detail: 'be active' }],
+  deliverables: [{ descriptionId: 3, dateAdded: '01/01/22', detail: 'did it work' }]
+};
+
+describe('Work Packages', () => {
+  beforeEach(() => {
+    prisma.work_Package.findMany = jest.fn();
+    prisma.work_Package.findUnique = jest.fn();
+  });
+
+  test('createWorkPackage fails if WBS number does not represent a project', async () => {
```

Comment:
you need to mock every prisma call that happens before the call to the endpoint finishes. For both of these, that means mocking `prisma.user.findUnique` and `prisma.change_Request.findUnique`. see the existing test files for how to do that

Reply:
Do we have to mock for prisma.work_Package.findUnique and prisma.work_Package.findAll? Also, would prisma.user.findUnique and prisma.change_Request.findUnique basically involve creating a user and creating a change request and then running jest.SpyOn() on both of them?

Reply:
1. in this case no you don't have to mock those because if the code is working as intended then the endpoint will fail before it even gets to those (returning the 400)
2. yeah pretty much. you can copy how that's done in the other files. it's kinda work intensive though so we should definitely in the future find a way to make it easier (probably by using pre made sample data that we can just import)

Reply:
@anthonybernardi Do I just test whether the two methods are called or is there more I have to do?

Reply:
what two methods? the title of the test is what you're testing. For example, if you pass in 1 for work package number, it should fail right. That's already the case, we're just mocking any prisma calls along the way so that we aren't relying on prisma being in any certain state.

Reply:
`prisma.user.findUnique` and `prisma.change_Request.findUnique` are the two methods, how would I make the mock to test it? ig I'm just kinda confused about how the mocks in the other test files work?